enhancements/dns/externaldns-operator-aws-assume-role.md

```diff
@@ -0,0 +1,301 @@
+
+In a future version of ExternalDNS Operator's API, we will resolve the requirement for the `Credentials` field through a
+more appropriate solution as outlined in the alternative [Bumping the API](#bumping-the-api).
+
+### Risks and Mitigations
+
+#### Security
+
+In terms of security concerns, the IAM role is not considered sensitive because AWS incorporates security measures to
+limit the usage of a role ARN to a particular AWS account.
+
+#### Lack of STS Support
+
+Another drawback for this feature is that the ExternalDNS Operator currently lacks support for the AWS Security Token
+Service (STS). Given that STS is not supported, and the majority of ROSA (Red Hat OpenShift Service on AWS) customers
+rely on STS, this shared VPC feature is not usable for most managed OpenShift users. The addition of STS support for
+ExternalDNS Operator is currently under evaluation. However, until that is completed, the decision has been made to move
+forward with the implementation of shared VPC support.
+
+### Drawbacks
+
+A minor drawback is the added complexity in supporting cross-account DNS hosted zones. This complexity involves handling
+AWS resource sharing, managing IAM policies and roles, and understanding shared VPC architecture.
+
+## Design Details
+
+### Open Questions
+
+N/A
+
+### Test Plan
+
+We will add E2E tests for the ExternalDNS Operator utilizing two AWS CI accounts to create DNS records in one (Account
+A) and install the cluster in the other (Account B). This E2E test will leverage existing steps in the step registry
+as created by [AWS Shared VPC with Cross-account DNS Zones](/enhancements/installer/aws-cross-account-dns-zone.md) such
+as [ipi-aws-pre-shared-vpc-phz](https://steps.ci.openshift.org/chain/ipi-aws-pre-shared-vpc-phz) and [ipi-aws-post-shared-vpc-phz](https://steps.ci.openshift.org/chain/ipi-aws-post-shared-vpc-phz).
+These steps set up a shared VPC cluster with access to a cross-account DNS private hosted zone. However, we may choose to use
+a public hosted zone instead, as there is no functional distinction between the two, and it simplifies the E2E testing
+process.
+
+### Graduation Criteria
+
+This update is an extension of the existing v1beta1 ExternalDNS Operator API and does not impact the graduation
+criteria for the v1beta1 API.
+
+The ExternalDNS Operator updates to support Shared VPC are out of payload and are therefore not aligned with
+OpenShift releases. Initially, the plan was to release this enhancement alongside of OCP 4.14. However, because
+ExternalDNS Operator lacks STS support (see [Lack of STS Support](#lack-of-sts-support)), the decision was made to
+complete implementation of this feature, but postpone the release until STS support is available.
+
+There is no requirement for backporting this feature, as outlined in [AWS Shared VPC with Cross-account DNS Zones](/enhancements/installer/aws-cross-account-dns-zone.md).
```

Comment:
The phrasing here is ambiguous.  

* "'AWS Shared VPC with Cross-account DNS Zones' outlines that there **is no requirement** for backporting this feature."  (This is how I interpreted the statement initially.)
* "'AWS Shared VPC with Cross-account DNS Zones' outlines that there **is a requirement** for backporting this feature in OCP, but there **is no such requirement** for ExternalDNS Operator."  (This is what I think you must actually mean.)

Anyway, the reason why we are not backporting the feature in ExternalDNS Operator is that you can run a newer EDNSO version on an older OCP cluster, right?  It would be useful to state that explicitly.